From d39421c8676ca3d2cfe82ce5f90c5cf506a1e861 Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hauke.mehrtens@intel.com>
Date: Wed, 31 Jul 2019 15:22:46 +0200
Subject: [PATCH] PONRTSYS-4774: Add padding also on the vUNI interface

When the destination MAC was already learned by the GSWIP the packets will take
the HW path on the bridge between the vUNI(PON) port and the pmapper. We will
not add any additional padding on this HW path and the packets will be dropped
somewhere. To make sure the padding is added already add the padding on
the vUNI driver.
---
 drivers/net/ethernet/lantiq/intel_pon_hgu_vuni.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/net/ethernet/lantiq/intel_pon_hgu_vuni.c b/drivers/net/ethernet/lantiq/intel_pon_hgu_vuni.c
index 7219092fd2f7..99e907229206 100644
--- a/drivers/net/ethernet/lantiq/intel_pon_hgu_vuni.c
+++ b/drivers/net/ethernet/lantiq/intel_pon_hgu_vuni.c
@@ -281,8 +281,14 @@ static int vuni_xmit(struct sk_buff *skb, struct net_device *dev)
 	int len;
 	struct net_device *vuni_dev;
 
+	if (skb_put_padto(skb, ETH_ZLEN)) {
+		priv->stats.tx_dropped++;
+		return NETDEV_TX_OK;
+	}
+
 	if (strcmp(dev->name, eth_dev[priv->id][0]->name) != 0) {
 		/*drop the packet*/
+		priv->stats.tx_dropped++;
 		kfree_skb(skb);
 		return 0;
 	}
